### ISSUE TITLE
Lexical Editor: Link and AutoLink not converting to markdown with lexical $convertToMarkdownString

### ISSUE THREAD

**Issue by shawnvogt** [payloadcms/payload#6507]
### Link to reproduction

https://github.com/TMRRWinc/PayloadLexicalLinkConvertToMarkdownIssue

### Describe the Bug

When using the $**convertToMarkdownString_function** from **@lexical/markdown**, the **Link** and **AutoLink** nodes do not correctly convert. 

expected output ```[link](https://payloadcms.com)``` actual output ```link```

if the Link and AutoLink nodes from _@lexical/link_ are subbed into the headless editor nodes, the the actual output changes to ```[link](undefined)```

### To Reproduce

run the "local API example" test in test/_community/int.spec.ts of the [linked reproduction](https://github.com/TMRRWinc/PayloadLexicalLinkConvertToMarkdownIssue/blob/main/packages/payload/package.json)

A richtext field with lexical editor was added to the posts collection. The richtext field has a field hook to convert the lexical state to markdown and attach it to context.  A collection AfterChangeHook adds the markdown value to the returned document to make the markdown value available to test against.

OR (follow the steps from the [docs ](https://payloadcms.com/docs/rich-text/lexical#lexical-markdown) and then try to convert a link)
###  Lexical => Markdown
Export content from the Lexical editor into Markdown format using these steps:

1. Import your current editor state into the headless editor.
2. Convert and fetch the resulting markdown string.

Here's the code for it:
```ts
import { $convertToMarkdownString } from '@lexical/markdown'
import { sanitizeEditorConfig } from '@payloadcms/richtext-lexical'
import type { SerializedEditorState } from "lexical"

const yourSanitizedEditorConfig = sanitizeEditorConfig(yourEditorConfig) // <= your editor config here
const yourEditorState: SerializedEditorState // <= your current editor state here

// Import editor state into your headless editor
try {
  headlessEditor.setEditorState(headlessEditor.parseEditorState(yourEditorState)) // This should commit the editor state immediately
} catch (e) {
  logger.error({ err: e }, 'ERROR parsing editor state')
}

// Export to markdown
let markdown: string
headlessEditor.getEditorState().read(() => {
  markdown = $convertToMarkdownString(yourSanitizedEditorConfig?.features?.markdownTransformers)
})
```

### Payload Version

2.18.3

### Adapters and Plugins

db-mongodb, @payload/richtext-lexical, @lexical/headless, @lexical/markdown

**Comment by AlessioGr** [comment]
Thank you for the detailed and helpful issue! Will be fixed in lexical 3.0.0-beta.37 :)

**Comment by AlessioGr** [comment]
Fixed by https://github.com/payloadcms/payload/pull/6543
